lib/spack/spack/build_systems/cmake.py

```diff
@@ -459,8 +459,11 @@ def cmake(
         """Runs ``cmake`` in the build directory"""
 
         # skip cmake phase if it is an incremental develop build
-        if spec.is_develop and os.path.isfile(
-            os.path.join(self.build_directory, "CMakeCache.txt")
+        # These are the files that will re-run CMake that are generated from a successful
+        # configure step
+        if spec.is_develop and (
+            os.path.isfile(os.path.join(self.build_directory, "Makefile"))
+            or os.path.isfile(os.path.join(self.build_directory, "ninja.build"))
         ):
```

Comment:
I'd feel a little more comfortable if we searched for one or the other based on the value of `_extract_primary_generator(self.generator)` rather than either/or because I don't know whether it's possible to have one be vestigial for some reason. I'll bow to your judgment on that, though.

Reply:
Seems unlikely, but this is also a reasonable request IMHO. I'll give it a stab.

Reply:
`generator` includes primary and possible secondary components, and you only care about the primary. That's why I suggested using `_extract_primary_generator(self.generator)` It should be as simple as `primary_generator = _extract_primary_generator(self.generator)` and then replacing `self.generator` with `primary_generator` in the conditional.

Reply:
I don't think I fully understand the nuance of what you're hitting at here. Apologies, could you expand on your concern of one being vestigial?  I don't really understand the concern. Are you concerned on if there is a Makefile when we use ninja for example? I removed the or so now an exact match is required. This logic was copied from the build phase so my understanding is the exact artifact of the appropriate configure has to exist to skip configuration.